WordPress/gutenberg
# Scrollable: Convert component to TypeScript

## What?
Converts the `Scrollable` component to TypeScript.

## Why?
Part of the @wordpress/components's TypeScript migration (https://github.com/WordPress/gutenberg/issues/35744).

## How?
- Refactors the current `Scrollable` to TypeScript.
- Refactors the tests to TypeScript.
- Refactors the storybook to TypeScript and removed knobs.

## Testing Instructions
1. Run npm run dev and ensure no typing errors
2. Fire up the Storybook and confirm the `Scrollable` continues to function as expected
